code/modules/mob/living/inventory.dm

```diff
@@ -122,17 +122,30 @@
 		for(var/slot in get_held_item_slots())
 			if(slot in old_slots)
 				LAZYSET(_inventory_slots, slot, old_slots[slot])
+				old_slots -= slot
```

Comment:
It might, yeah. I'm not clear on the intended behavior - do we want to keep the held item slots by default and only replace them if there are new held item slots passed in ``new_slots`` (in which case the current behavior should work ok), or do we always want to keep the held item slots the same regardless of what is passed in ``new_slots``?